### PR TITLE
Implement  PSKT(Partially Signed Kaspa Transaction)

### PR DESCRIPTION
**Pull request by biryukovmaxim**
**Description:**

This PR introduces the implementation of Kaspa's version of Partially Signed Transaction (PST) handling, inspired by Bitcoin's PSBT (Partially Signed Bitcoin Transaction). The implementation includes various roles and functionalities necessary for creating, updating, signing, combining, finalizing, and extracting PSTs.

### Key Features

- **Format and Constraint Operations**: The implementation focuses on the format and constraint operations for PSTs.
- **Integration**: Designed to be integrated into a wallet framework or any other wallet implementation.
- **API Clarity**: The API is clear for integration, exposing the PST and waiting for signatures in the case of the Signer role.
- **No Private Key Handling**: It does not accept private keys, nor does it handle script details like multisig or HTLC directly.
- **Signature Handling**: In the Finalizer role, it accepts a callback that constructs the final signature.
- **Example Provided**: A comprehensive example is included demonstrating how partial signatures, redeem scripts, and mixing operations are handled.

### CODE REVIEW

**Comment by coderofstuff** [comment]
Upkeep stuff:
- please update the PR title to be something clearer - to me it's kaspa's version of PSBT but it might not be immediately clear to anyone else 
- please add a description for what this PR is about and how to test it

**Review by coderofstuff — COMMENTED** [review]
Just some initial comments first

File: crypto/txscript/src/standard/multisig.rs [hunk 1]
```diff
@@ -130,18 +130,17 @@ mod tests {
...
         };
-
         let tx = Transaction::new(
             0,
             vec![TransactionInput {
                 previous_outpoint: TransactionOutpoint { transaction_id: prev_tx_id, index: 0 },
                 signature_script: vec![],
-                sequence: 0,
-                sig_op_count: 4,
+                sequence: u64::MAX,
```
[the remark this answers is not in the document]
Comment (biryukovmaxim) [reply crypto/txscript/src/standard/multisig.rs:138]: unrelated changes that does nothing with test. i ll rolback it

File: wallet/pskt/src/role.rs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+pub enum Creator {}
```
[the remark this answers is not in the document]
Comment (biryukovmaxim) [reply wallet/pskt/src/role.rs:3]: added with [commit](https://github.com/kaspanet/rusty-kaspa/pull/481/commits/9bda6e43df59e5f98ac1e2882991e997170d8151)

File: wallet/pskt/src/global.rs [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    /// The version numbers are not the same.
+    VersionMismatch {
+        /// Attempted to combine a PBST with `this` version.
+        this: Version,
+        /// Into a PBST with `that` version.
+        that: Version,
+    },
+    #[error("The transaction version numbers are not the same")]
+    TxVersionMismatch {
+        /// Attempted to combine a PBST with `this` tx version.
```
[the remark this answers is not in the document]
Comment (biryukovmaxim) [reply wallet/pskt/src/global.rs]: done

**Comment by biryukovmaxim** [comment]
> * please add a description for what this PR is about and how to test it

Description is added. 
Regarding to test - there is no clear answer. I guess it should be integrated to wallet framework or any other wallet implemtation.
It mostly just  limited set of operations according to roles and format of the pskt

**Review by coderofstuff — APPROVED** [review]

**Pull request merged**